src/torcontrol.cpp

```diff
@@ -323,6 +376,10 @@ static std::pair<bool,std::string> ReadBinaryFile(const fs::path &filename, size
     char buffer[128];
     size_t n;
     while ((n=fread(buffer, 1, sizeof(buffer), f)) > 0) {
+        // Check for reading errors so we don't return any data if we couldn't
+        // read the entire file (or up to maxsize)
+        if (ferror(f))
+            return std::make_pair(false,"");
```

Comment:
Interesting, I didn't know that, I though that only the EOF case would cause it to return a smaller amount larger or equal to zero (which according to the manual page is the case for `read`, but apparently not `fread`).